src/components/ReportingConfig/index.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Collapsible, Icon } from '@edx/paragon';
+import LMSApiService from '../../data/services/LmsApiService';
+import ReportingConfigForm from './ReportingConfigForm';
+import { camelCaseObject, snakeCaseFormData } from '../../utils';
+import LoadingMessage from '../LoadingMessage';
+import ErrorPage from '../ErrorPage';
+
+class ReportingConfig extends React.Component {
+  state = {
+    loading: true,
+    reportingConfigs: [],
+    error: undefined,
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  newConfigFormRef = React.createRef();
+
+  componentDidMount() {
+    LMSApiService.fetchReportingConfigs(this.props.enterpriseId)
+      .then(response => this.setState({
+        reportingConfigs: response.data.results,
+        loading: false,
+      }))
+      .catch(error => this.setState({ error }));
+  }
+
+  /**
+   * Creates a new reporting configuration, then updates this list with the response.
+   * Returns if there is an error.
+   * @param {FormData} formData
+   */
+  createConfig = async (formData) => {
+    //  snake_case the data before sending it to the backend
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      const response = await LMSApiService.postNewReportingConfig(transformedData);
+      this.setState(prevState => ({
+        reportingConfigs: [
+          ...prevState.reportingConfigs,
+          response.data,
+        ],
+      }));
+      this.newConfigFormRef.current.close();
+      return undefined;
```

Comment:
Not saying it's incorrect, I just don't understand the use of it, and it think it's been talked about recently (regarding `void`)?

Reply:
This was done because eslint wants a consistent return since it's returning an error. I don't know if this is the right choice, I'm open to suggestions (eslint doesn't like void either).